touchforms/formplayer/static/formplayer/script/entrycontrols_full.js

```diff
@@ -1,785 +1,584 @@
+    this.prevalidate = function(q) {
+        return true;
+    }
+    this.clear = function() {
+        this.answer(null);
     }
-    var retval = basefunc.apply(self, arguments);
-    self._parent[funcName].push(basefunc);
-    return retval;
-  }
 }
 
-/* TO USE:
- *
- * early in the constructor of the child class, call:
- *   inherit(this, new ParentClass(...));
- * this is akin to calling parent(...); inside a java constructor
- *
- * this call will load all variables and functions from the parent class
- * into this class.
- *
- * add additional variables and methods after the inherit() call. to
- * overload a method in the parent class, simply redefine the function
- * in this class
- *
- * to call a parent method explicitly, do:
- *   this.parent('someMethod')(args);
- * this is akin to calling parent.someMethod(args); in java
+/**
+ * An entry that represent a question label.
  */
+function InfoEntry(question, options) {
+    Entry.call(self, question, options);
+    this.answer = question.answer;
+    this.templateType = 'blank'
+};
 
-/* ============================= */
-
+InfoEntry.prototype = Object.create(Entry.prototype);
+InfoEntry.prototype.constructor = Entry;
 
-function Entry () {
-  this.help = function () {
-    showError(activeQuestion["help"] || "There is no help text for this question.");
-  }
-
-  this.clear = function () {
-    this.setAnswer(null, true);
-  }
-
-  this.onShow = function() { }
+/**
+ * The entry used when we have an unidentified entry
+ */
+function UnsupportedEntry(question, options) {
+    Entry.call(self, question, options);
+    this.templateType = 'unsupported'
+    this.answer = null;
 }
+UnsupportedEntry.prototype = Object.create(Entry.prototype);
+UnsupportedEntry.prototype.constructor = Entry;
 
-function SimpleEntry () {
-  inherit(this, new Entry());
+/**
+ * The entry that represents a free text input
+ * TODO: Handle prose option
+ */
+function FreeTextEntry(question, options) {
+    var self = this;
+    Entry.call(self, question, options);
+    self.templateType = 'str';
+    self.domain = question.domain ? question.domain() : 'full';
+    self.lengthLimit = options.lengthLimit || 10000;
+    self.prose = question.domain_meta ? question.domain_meta().longtext : false;
+
+    self.answer = question.answer;
+    self.answer.subscribe(function(newValue) {
+        self.question.onchange()
+    });
 
-  this.shortcuts = [];
+    this.prevalidate = function() {
+        var errmsg = this._prevalidate(self.answer());
+        if (errmsg) {
+            self.question.showError(errmsg);
+            return false;
+        }
+        return true;
+    }
 
-  this.prevalidate = function (q) {
-    return true;
-  }
+    this._prevalidate = function(raw) {
+        return null;
+    }
 
-  this.destroy = function () {
-    for (var i = 0; i < this.shortcuts.length; i++) {
-      shortcut.remove(this.shortcuts[i]);
+    this.domainText = function() {
```

Comment:
terrible name, but i think this is supposed to be the type of entry you are filling out. planning to change to something like human readable entry